src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp

```diff
@@ -181,6 +181,19 @@ class TypeObjectRegistry : public ITypeObjectRegistry
             const std::string& type_name,
             TypeInformation& type_information);
 
+    /**
+     * @brief Get the type dependencies of the given direct hash type identifiers.
+     *
+     * @param[in] type_identifiers Sequence with the queried direct hash TypeIdentifiers.
+     * @param[in out] type_dependencies Unordered set of TypeIdentifiers with related TypeObject serialized size.
+     * @return ReturnCode_t RETCODE_OK if the operation is successful.
+     *                      RETCODE_NO_DATA if any given TypeIdentifier is unknown to the registry.
+     *                      RETCODE_BAD_PARAMETER if any given TypeIdentifier is not a direct hash.
+     */
+    ReturnCode_t get_direct_hash_type_dependencies(
+            const TypeIdentifierSeq& type_identifiers,
+            std::unordered_set<TypeIdentfierWithSize>& type_dependencies);
+
```

Comment:
AS I suggested in the source file. I suggest that the public API remains unchanged. And the thing is that the previous implementation is now a private method with a new name. Please, ensure that the implementation method is private.

Reply:
moved i n[d3245e1](https://github.com/eProsima/Fast-DDS/pull/4496/commits/d3245e16dd687c62ce397be8018c718828d2bc6e)